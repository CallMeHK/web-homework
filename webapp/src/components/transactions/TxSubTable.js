import React from 'react'
import { string, bool, shape } from 'prop-types'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableContainer from '@material-ui/core/TableContainer'
import TableRow from '@material-ui/core/TableRow'
import Paper from '@material-ui/core/Paper'

export function TxSubTable ({ credit, debit, description, merchant }) {
  return (
    <TableContainer component={Paper}>
      <Table aria-label='collapsible table'>
        <TableBody>
          <TableRow>
            <TableCell>Purchase Type</TableCell>
            <TableCell data-testid="payment" align='right'>{credit ? 'Credit' : debit ? 'Debit' : 'Other'}</TableCell>
          </TableRow>
          <TableRow>
            <TableCell>Transaction Description</TableCell>
            <TableCell data-testid="transactionDescription" align='right'>{description}</TableCell>
          </TableRow>
          <TableRow>
            <TableCell>Merchant Description</TableCell>
            <TableCell data-testid="merchantDescription" align='right'>{merchant.description}</TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </TableContainer>
  )
}

TxSubTable.propTypes = {
  credit: bool.isRequired,
  debit: bool.isRequired,
  description: string.isRequired,
  merchant: shape({
    description: string.isRequired
  }).isRequired
}
