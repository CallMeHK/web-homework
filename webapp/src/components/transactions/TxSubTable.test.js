import React from 'react'
import renderer from 'react-test-renderer'
import { TxSubTable } from './TxSubTable'

describe('TxSubTable', () => {
  describe('Other payment', () => {
    it('should render other', () => {
      const tree = renderer
        .create(
          <TxSubTable
            credit={false}
            debit={false}
            description="transactionDescription"
            merchant={{ description: 'merchantDescription' }}
          />
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Credit payment', () => {
    it('should render Credit', () => {
      const tree = renderer
        .create(
          <TxSubTable
            credit={true}
            debit={false}
            description="transactionDescription"
            merchant={{ description: 'merchantDescription' }}
          />
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Debit payment', () => {
    it('should render Debit', () => {
      const tree = renderer
        .create(
          <TxSubTable
            credit={false}
            debit={true}
            description="transactionDescription"
            merchant={{ description: 'merchantDescription' }}
          />
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
