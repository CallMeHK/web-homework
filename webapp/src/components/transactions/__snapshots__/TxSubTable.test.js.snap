// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxSubTable Credit payment should render Credit 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="collapsible table"
    className="MuiTable-root"
    role={null}
  >
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Purchase Type
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          data-testid="payment"
        >
          Credit
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Transaction Description
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          data-testid="transactionDescription"
        >
          transactionDescription
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Merchant Description
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          data-testid="merchantDescription"
        >
          merchantDescription
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TxSubTable Debit payment should render Debit 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="collapsible table"
    className="MuiTable-root"
    role={null}
  >
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Purchase Type
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          data-testid="payment"
        >
          Debit
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Transaction Description
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          data-testid="transactionDescription"
        >
          transactionDescription
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Merchant Description
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          data-testid="merchantDescription"
        >
          merchantDescription
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TxSubTable Other payment should render other 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="collapsible table"
    className="MuiTable-root"
    role={null}
  >
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Purchase Type
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          data-testid="payment"
        >
          Other
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Transaction Description
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          data-testid="transactionDescription"
        >
          transactionDescription
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Merchant Description
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          data-testid="merchantDescription"
        >
          merchantDescription
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
